packages/cli/generators/controller/index.js

```diff
@@ -73,3 +148,90 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
     }
   }
 };
+
+/**
+ * Optional prompt to retrieve variables for artifacts.
+ * Not a member of the class because Yeoman will always call Generator
+ * prototype functions! Use .apply or .call with this function!
+ */
+function promptArtifactCrudVars() {
+  const self = this;
+  let modelList = [];
+  let repositoryList = [];
+  return utils
+    .getArtifactList(self.artifactInfo.modelDir, 'model')
+    .then(list => {
+      if (_.isEmpty(list)) {
+        return Promise.reject(
```

Comment:
We should call this.exit(error) to gracefully exit.

Reply:
It seems that existence of `models` is required for `lb4 controller` to be useful. We probably need to add `lb4 model` soon.